src/wallet/rpcwallet.cpp

```diff
@@ -121,6 +123,14 @@ void EnsureWalletIsUnlocked(const CWallet* pwallet)
     }
 }
 
+WalletContext& EnsureWalletContext(const util::Ref& context)
+{
+    if (!context.Has<WalletContext>()) {
```

Comment:
Why not `assert(context.Has<WalletContext>())` for now?

Reply:
Is it possible to hit this exception?

Reply:
assert can not be used in rpc code. See

```
test/lint/lint-assertions.sh:# Macro CHECK_NONFATAL(condition) should be used instead of assert for RPC code, where it